mm2src/coins/z_coin.rs

```diff
@@ -763,9 +766,23 @@ impl<'a> UtxoCoinWithIguanaPrivKeyBuilder for ZCoinBuilder<'a> {
         .expect("DEX_FEE_Z_ADDR is a valid z-address")
         .expect("DEX_FEE_Z_ADDR is a valid z-address");
 
-        let z_tx_prover = async_blocking(LocalTxProver::with_default_location)
-            .await
-            .or_mm_err(|| ZCoinBuildError::ZCashParamsNotFound)?;
+        let z_tx_prover = match &self.z_coin_params.zcash_params_path {
+            None => async_blocking(LocalTxProver::with_default_location)
+                .await
+                .or_mm_err(|| ZCoinBuildError::ZCashParamsNotFound)?,
+            Some(file_path) => {
+                let path = PathBuf::from(file_path);
+                let (spend_path, output_path) = if path.exists() {
+                    (path.join(SAPLING_SPEND_NAME), path.join(SAPLING_OUTPUT_NAME))
+                } else {
+                    return MmError::err(ZCoinBuildError::ZCashParamsNotFound);
+                };
+                if !(spend_path.exists() && output_path.exists()) {
+                    return MmError::err(ZCoinBuildError::ZCashParamsNotFound);
+                }
+                LocalTxProver::new(&spend_path, &output_path)
```

Comment:
As there are FS operations and `LocalTxProver::new` can take significant time to load, please wrap this entire code block into `async_blocking`.

Reply:
Oh, I see.
Done